profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSamplerLoop.cpp

```diff
@@ -580,7 +579,7 @@ void StackSamplerLoop::PersistStackSnapshotResults(
         rawSample.Timestamp = pSnapshotResult->GetUnixTimeUtc();
         rawSample.LocalRootSpanId = pSnapshotResult->GetLocalRootSpanId();
         rawSample.SpanId = pSnapshotResult->GetSpanId();
-        rawSample.AppDomainId = pSnapshotResult->GetAppDomainId();
```

Comment:
Even if today the "current" thread's appdomain is retrieved AFTER the thread gets resumed, see if we could get it BEFORE the stack is walked and then passed to PersistStackSnapshotResults. This applied to Windows only because there is only one appdomain on linux